enhancements/ingress/ingress-dashboard.md

```diff
@@ -0,0 +1,199 @@
+---
+title: ingress-dashboard
+authors:
+  - "@jotak"
+reviewers:
+  - "@Miciah"
+  - "@candita"
+  - "@OlivierCazade"
+approvers:
+  - "@Miciah"
+  - "@candita"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-06-28
+last-updated: 2023-06-28
+tracking-link:
+  - "https://issues.redhat.com/browse/OCPSTRAT-139"
+  - "https://issues.redhat.com/browse/NETOBSERV-1052"
+see-also:
+replaces:
+superseded-by:
+---
+
+# Ingress dashboard creation for the OpenShift Console
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`.
+- [ ] Design details are appropriately documented from clear requirements.
+- [ ] Test plan is defined.
+- [ ] graduation criteria for dev preview, tech preview, GA
+- [ ] User-Facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/).
+
+## Summary
+
+Add a new dashboard in the OpenShift Console (in menu “Observe” > “Dashboards”), dedicated to metrics related to Ingress.
+
+Ingress components, such as HAProxy, already provide some metrics that are exposed and collected by Prometheus / Cluster Monitoring. Administrators should be able to get a consolidated view, using a subset of these metrics, to get a quick overview of the cluster state. This enhancement proposal is part of a wider initiative to improve the observability of networking components (cf https://issues.redhat.com/browse/OCPSTRAT-139).
+
+## Motivation
+
+While ingress related metrics already exist and are accessible in the OpenShift Console (via the menu “Observe” > Metrics”), there is no consolidated view presenting a summary of them. There are existing dashboards in other areas (such as etcd, compute resources, etc.), but networking today is less represented there, despite its importance for monitoring and troubleshooting.
+
+In addition, product management has shown interest in providing and making more visible some cluster-wide statistics such as the number of routes and shards in use.
+
+This [request for enhancement](https://issues.redhat.com/browse/NE-1059) is also considered here, although it will not be entirely fulfilled as we don't intend here to create any new metric.
+
+### Goals
+
+1. Review and select a subset of the available metrics that should be included in the new dashboard. This should ideally be done with the help of the NetEdge team expertise and/or SREs.
```

Comment:
This should be a part of the EP, a list of the required metrics.

Reply:
I'm not sure that an _exact_ list of metrics needs to be identified in the EP, at least initially. It would be better to have the mechanics wired up so that we can iteratively and incrementally add new metrics once we understand what would be genuinely useful to a cluster administrator/operator. Finding useful metrics could come from taking an existing Grafana dashboard, looking at HAProxy's built-in dashboard, working with SREs, et al.

Reply:
I did provide a writeup that listed our available metrics, and a proposal we used when we were talking to SRE about this over a year ago.